tests/providers/amazon/aws/hooks/test_batch_waiters.py

```diff
@@ -0,0 +1,486 @@
+
+from airflow import AirflowException
+from airflow.providers.amazon.aws.hooks.batch_waiters import AwsBatchWaiters
+
+# Use dummy AWS credentials
+AWS_REGION = "eu-west-1"
+AWS_ACCESS_KEY_ID = "airflow_dummy_key"
+AWS_SECRET_ACCESS_KEY = "airflow_dummy_secret"
+
+
+@pytest.fixture(scope="module")
+def aws_region():
+    return AWS_REGION
+
+
+@pytest.fixture(scope="module")
+def job_queue_name():
+    return "moto_test_job_queue"
+
+
+@pytest.fixture(scope="module")
+def job_definition_name():
+    return "moto_test_job_definition"
+
+
+#
+# AWS Clients
+#
+
+
+class AwsClients(NamedTuple):
+    batch: "botocore.client.Batch"
+    ec2: "botocore.client.EC2"
+    ecs: "botocore.client.ECS"
+    iam: "botocore.client.IAM"
+    log: "botocore.client.CloudWatchLogs"
+
+
+@pytest.yield_fixture(scope="module")
+def batch_client(aws_region):
+    with mock_batch():
+        yield boto3.client("batch", region_name=aws_region)
+
+
+@pytest.yield_fixture(scope="module")
+def ec2_client(aws_region):
+    with mock_ec2():
+        yield boto3.client("ec2", region_name=aws_region)
+
+
+@pytest.yield_fixture(scope="module")
+def ecs_client(aws_region):
+    with mock_ecs():
+        yield boto3.client("ecs", region_name=aws_region)
+
+
+@pytest.yield_fixture(scope="module")
+def iam_client(aws_region):
+    with mock_iam():
+        yield boto3.client("iam", region_name=aws_region)
+
+
+@pytest.yield_fixture(scope="module")
+def logs_client(aws_region):
+    with mock_logs():
+        yield boto3.client("logs", region_name=aws_region)
+
+
+@pytest.fixture(scope="module")
+def aws_clients(batch_client, ec2_client, ecs_client, iam_client, logs_client):
+    return AwsClients(
+        batch=batch_client, ec2=ec2_client, ecs=ecs_client, iam=iam_client, log=logs_client
+    )
+
+
+#
+# Batch Infrastructure
+#
+
+
+class Infrastructure:
+    aws_region: str
+    aws_clients: AwsClients
+    vpc_id: Optional[str] = None
+    subnet_id: Optional[str] = None
+    security_group_id: Optional[str] = None
+    iam_arn: Optional[str] = None
+    compute_env_name: Optional[str] = None
+    compute_env_arn: Optional[str] = None
+    job_queue_name: Optional[str] = None
+    job_queue_arn: Optional[str] = None
+    job_definition_name: Optional[str] = None
+    job_definition_arn: Optional[str] = None
+
+
+def batch_infrastructure(
+    aws_clients: AwsClients, aws_region: str, job_queue_name: str, job_definition_name: str
+) -> Infrastructure:
+    """
+    This function is not a fixture so that tests can pass the AWS clients to it and then
+    continue to use the infrastructure created by it while the client fixtures are in-tact for
+    the duration of a test.
+
+    :param aws_clients:
+    :type aws_clients: AwsClients
+
+    :param aws_region:
+    :type aws_region: str
+
+    :param job_queue_name:
+    :type job_queue_name: str
+
+    :param job_definition_name:
+    :type job_definition_name: str
```

Comment:
We should complete the documentation here.

Reply:
This is just test code in one test file;  AFAIK, this is not public and so long as the linters are OK with it, it should suffice.

Reply:
You are right. Let's remove them then. 👍

Reply:
Removed ✔️